### ISSUE TITLE
API only returning fraction of vulnerabilities

### ISSUE THREAD

**Issue by ChrisSpraaklab** [wazuh/wazuh#26468]
|Wazuh version|Component|
|---|---|
| 4.9.1-1 | API |

## Description
When calling the API to get a list of vulnerabilities for a specific agent, only a handful are returned where I know that agent has a lot more vulnerabilities (>1000) from the dashboard. For some agents, even no vulnerabilities are returned when again a lot are showing in the dashboard. The vulnerability detection dashboard also shows some confusing mixups of agent ids and corresponding names for some agents. When I choose an agent (using the button "explore agent"), vulnerabilities for another agent are shown. The same mixups occur in the api outputs.

## Checks

The following elements have been updated or reviewed (should also be checked if no modification is required):
- [ ] Tests (unit tests, API integration tests).
- [ ] Changelog.
- [ ] Documentation.
- [ ] Integration test mapping (using `api/test/integration/mapping/_test_mapping.py`).

**Comment by SirStephanikus** [comment]
Here, everything works fine ... no mix ups or other strange behavior.

Could you please give us the API call ?

**Comment by ChrisSpraaklab** [comment]
```
curl -k -u admin:admin -X GET "https://localhost:9200/wazuh-states-vulnerabilities-*/_search" -H 'Content-Type: application/json' -d '{
    "query": {
        "match": {
            "agent.id": "001"
        }
    }
}'```

**Comment by SirStephanikus** [comment]
```
{
  "took": 2,
  "timed_out": false,
  "_shards": {
    "total": 1,
    "successful": 1,
    "skipped": 0,
    "failed": 0
  },
  "hits": {
    "total": {
      "value": 1757,
      "relation": "eq"
    },
```

The above output is from an old linux vm ... works.

Then, my MAC:
```

{
  "took": 2,
  "timed_out": false,
  "_shards": {
    "total": 1,
    "successful": 1,
    "skipped": 0,
    "failed": 0
  },
  "hits": {
    "total": {
      "value": 43,
      "relation": "eq"
    },
```
Looks valid too.

If I change the agent inside the dashboards, everything runs fine.

Check you logs in debug mode, perhaps you messed something up???


Try this too:

Inside your dev-console of your indexer:

```
GET wazuh-states-vulnerabilities-*/_search?size=5000&_source_includes=agent.name,package.name,package.version,package.description,vulnerability.severity,vulnerability.score.base,vulnerability.description,vulnerability.id,vulnerability.published_at
{
"query": {
  "bool": {
    "must": [
      {
        "match": {"agent.id": "001"}
      }]
    }
  }
}
```

**Comment by ChrisSpraaklab** [comment]
The size parameter fixed the missing vulnerabilities. The mixup of agent ids & names however still persists. This problem seems to arise somewhere else within my installation, maybe the manager? I recently performed a fresh install of the manager to resolve some other issues.

**Comment by SirStephanikus** [comment]
> The size parameter fixed the missing vulnerabilities. The mixup of agent ids & names however still persists. This problem seems to arise somewhere else within my installation, maybe the manager? I recently performed a fresh install of the manager to resolve some other issues.

Thanks for your feedback and nice to see that I helped you.

Well, if you re-installed your manager, but not your indexer ... it may be possible that there is some chaos inside?!?
The indexer is your database (opensearch is a non-relational database). 

There are many steps to solve the issues, but depending on the real root cause it may be better to do a clean fresh install ... I've no idea how many agents are involved and if it's a productive environment.

Are your Agent ID messed up in the other Wazuh areas too (like Threat-Hunting etc.?)

**Comment by ChrisSpraaklab** [comment]
My bad, the fresh install I mentioned was of all components, not just the manager. The agent ids are correct in the threath-hunting section.

**Comment by SirStephanikus** [comment]
I can only do guess work here, but it sounds that there are still old pieces from the prior setup left on your disk in /var/ossec/queue/

Wazuh is usually rock solid, always do anything in a test environment.

**Comment by ChrisSpraaklab** [comment]
I had the same suspicion, so I emptied that directory. This didn't solve it. I have now removed and added an affected agent a few times, where every time it would display vulnerabilities for a diferent agent that was not in use anymore. I think that these agent ids were still assigned to those old agents (with their respective vulnerabilities) somewhere. After doing this a few times I arrived to a range of agent ids that was never used before, which seems to fix it.  
Thanks for the assistance.
